crates/oxc_minifier/src/peephole/minimize_conditions.rs

```diff
@@ -353,130 +353,108 @@ impl<'a> PeepholeOptimizations {
-            if matches!(
-                test_expr.operator,
-                BinaryOperator::Inequality | BinaryOperator::StrictInequality
-            ) {
-                test_expr.operator = test_expr.operator.equality_inverse_operator().unwrap();
-                let test = ctx.ast.move_expression(&mut expr.test);
-                let consequent = ctx.ast.move_expression(&mut expr.consequent);
-                let alternate = ctx.ast.move_expression(&mut expr.alternate);
-                return Some(
-                    ctx.ast.expression_conditional(expr.span, test, alternate, consequent),
-                );
+            // "!a ? b : c" => "a ? c : b"
+            Expression::UnaryExpression(test_expr) => {
+                if test_expr.operator.is_not() {
+                    let test = ctx.ast.move_expression(&mut test_expr.argument);
+                    let consequent = ctx.ast.move_expression(&mut expr.alternate);
+                    let alternate = ctx.ast.move_expression(&mut expr.consequent);
+                    return Some(
+                        ctx.ast.expression_conditional(expr.span, test, consequent, alternate),
+                    );
+                }
             }
+            // "a ? a : b" => "a || b"
+            Expression::Identifier(id) => {
+                if let Expression::Identifier(id2) = &expr.consequent {
+                    if id.name == id2.name {
+                        return Some(ctx.ast.expression_logical(
+                            expr.span,
+                            ctx.ast.move_expression(&mut expr.test),
+                            LogicalOperator::Or,
+                            ctx.ast.move_expression(&mut expr.alternate),
+                        ));
+                    }
+                }
+                // "a ? b : a" => "a && b"
+                if let Expression::Identifier(id2) = &expr.alternate {
+                    if id.name == id2.name {
+                        return Some(ctx.ast.expression_logical(
+                            expr.span,
+                            ctx.ast.move_expression(&mut expr.test),
+                            LogicalOperator::And,
+                            ctx.ast.move_expression(&mut expr.consequent),
+                        ));
+                    }
+                }
+            }
+            _ => {}
         }
 
-        // TODO: `/* @__PURE__ */ a() ? b : b` -> `b`
-
-        // `a ? b : b` -> `a, b`
-        if expr.alternate.content_eq(&expr.consequent) {
-            let expressions = ctx.ast.vec_from_array([
-                ctx.ast.move_expression(&mut expr.test),
-                ctx.ast.move_expression(&mut expr.consequent),
-            ]);
-            return Some(ctx.ast.expression_sequence(expr.span, expressions));
-        }
-
-        // `a ? true : false` -> `!!a`
-        // `a ? false : true` -> `!a`
-        if let (
-            Expression::Identifier(_),
```

Comment:
The minsize reduction is probably due to the removal of this check, which is not present in the esbuild code.